jas/runner/jasrunner.go

```diff
@@ -31,34 +32,54 @@ func AddJasScannersTasks(securityParallelRunner *utils.SecurityParallelRunner, s
 	if !thirdPartyApplicabilityScan {
 		for _, module := range scanner.JFrogAppsConfig.Modules {
 			if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SecretsScan) {
 				log.Debug("Skipping secrets scan as requested by input...")
+			} else if configProfile != nil {
+				// This code section is related to CentralizedConfig integration in CI Next.
+				log.Debug(fmt.Sprintf("Using config profile '%s' to determine whether to run secrets scan...", configProfile.ProfileName))
+				if configProfile.Modules[0].ScanConfig.SecretsScannerConfig.EnableSecretsScan {
+					err = addModuleJasScanTask(jfrogappsconfig.Module{}, jasutils.Secrets, securityParallelRunner, runSecretsScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module, secretsScanType), errHandlerFunc)
+				} else {
+					log.Debug(fmt.Sprintf("Skipping secrets scan as requested by '%s' config profile...", configProfile.ProfileName))
+				}
 			} else if err = addModuleJasScanTask(module, jasutils.Secrets, securityParallelRunner, runSecretsScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module, secretsScanType), errHandlerFunc); err != nil {
 				return
 			}
 			if runAllScanners {
-				if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.IacScan) {
-					log.Debug("Skipping Iac scan as requested by input...")
-				} else if err = addModuleJasScanTask(module, jasutils.IaC, securityParallelRunner, runIacScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
-					return
+				if configProfile == nil {
+					if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.IacScan) {
+						log.Debug("Skipping Iac scan as requested by input...")
+					} else if err = addModuleJasScanTask(module, jasutils.IaC, securityParallelRunner, runIacScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
+						return
+					}
 				}
 				if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.SastScan) {
 					log.Debug("Skipping Sast scan as requested by input...")
+				} else if configProfile != nil {
+					log.Debug(fmt.Sprintf("Using config profile '%s' to determine whether to run Sast scan...", configProfile.ProfileName))
+					if configProfile.Modules[0].ScanConfig.SastScannerConfig.EnableSastScan {
+						err = addModuleJasScanTask(jfrogappsconfig.Module{}, jasutils.Sast, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc)
+					} else {
+						log.Debug(fmt.Sprintf("Skipping Sast scan as requested by '%s' config profile...", configProfile.ProfileName))
+					}
 				} else if err = addModuleJasScanTask(module, jasutils.Sast, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
 					return
 				}
 			}
```

Comment:
```suggestion
            if err = addJasScanTask(jasutils.Secrets, scansToPreform, configProfile, module, securityParallelRunner, runSecretsScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module, secretsScanType), errHandlerFunc); err != nil {
                return
            }
			if runAllScanners {
				if configProfile == nil {
					if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, utils.IacScan) {
						log.Debug("Skipping Iac scan as requested by input...")
					} else if err = addModuleJasScanTask(module, jasutils.IaC, securityParallelRunner, runIacScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
						return
					}
				}
                if err = addJasScanTask(jasutils.Sast, scansToPreform, configProfile, module, securityParallelRunner, runSastScan(securityParallelRunner, scanner, scanResults.ExtendedScanResults, module), errHandlerFunc); err != nil {
                    return
                }
			}
```
```
func addJasScanTask(scan utils.SubScanType, scansToPreform []utils.SubScanType, configProfile *services.ConfigProfile, module jfrogappsconfig.Module, securityParallelRunner *utils.SecurityParallelRunner, task parallel.TaskFunc, errHandlerFunc func(error)) (err error) {
    if len(scansToPreform) > 0 && !slices.Contains(scansToPreform, scan) {
		log.Debug(fmt.Sprintf("Skipping %s scan as requested by input..."), scan.String())
        return
    }
    if configProfile != nil {
        // This code section is related to CentralizedConfig integration in CI Next.
		log.Debug(fmt.Sprintf("Using config profile '%s' to determine whether to run %s scan...", configProfile.ProfileName), scan.String())
		if shouldRunScan(scan, configProfile) {
			err = addModuleJasScanTask(jfrogappsconfig.Module{}, scan, securityParallelRunner, task, errHandlerFunc)
		} else {
			log.Debug(fmt.Sprintf("Skipping %s scan as requested by '%s' config profile...", scan.String(), configProfile.ProfileName))
		}
        return
    }
    return addModuleJasScanTask(module, scan, securityParallelRunner, task, errHandlerFunc)
}

func shouldRunScan(scan utils.SubScanType, configProfile *services.ConfigProfile) bool {
    switch scan {
    case utils.Secrets:
    return configProfile.Modules[0].ScanConfig.SecretsScannerConfig.EnableSecretsScan
    case utils.Sast:
    return configProfile.Modules[0].ScanConfig.SastScannerConfig.EnableSastScan
    default:
        return false
    }
}
```

Reduce code duplications

Reply:
As we agreed, since this piece of code is going to change soon by https://github.com/jfrog/jfrog-cli-security/pull/96 , I will leave the code like this and not splitting to functions, and we will complete it in the proper way during the refactor.
Integration tests will be added to check this logic though